samples/tools/testbed/utils/download_humaneval.py

```diff
@@ -0,0 +1,67 @@
+#
+# Run this file to download the human_eval dataset, and create a corresponding testbed scenario:
+# (default: ../scenarios/human_eval_two_agents_gpt4.jsonl and ./scenarios/human_eval_two_agents_gpt35.jsonl)
+#
+
+import requests
+import gzip
+import io
+import json
+import os
+import base64
+
+
+script_path = os.path.realpath(__file__)
+script_name = os.path.basename(script_path)
+script_dir = os.path.dirname(script_path)
+
+# Directory where scenarios are stored
+scenarios_dir = os.path.realpath(os.path.join(script_dir, os.path.pardir, "scenarios"))
+print("Saving HumanEval scenarios to: " + scenarios_dir)
+
+
+# URL of the file to download
+url = "https://github.com/openai/human-eval/raw/master/data/HumanEval.jsonl.gz"
+
+# Send a HTTP request to the URL of the file
+response = requests.get(url)
+
+# Ensure we raise an error if the download failed
+response.raise_for_status()
+
+# Create a BytesIO object from the response content
+buffer = io.BytesIO(response.content)
+
+# Create a scenario file
+fh_gpt4 = open(os.path.join(scenarios_dir, "human_eval_two_agents_gpt4.jsonl"), "wt")
```

Comment:
use with open("test.jsonl", "wt") as f:

Reply:
That's not ideal in this case because I have two open file handles and I bounce between. Using "with" with two handles  requires two extra levels of nesting (unless I'm mistaken). Classic file handles work well for this. There's very little risk to forgetting to close the file, or having it dangle open on exception since it immediately closes and exits the process afterward anyway.

Reply:
I'll mark this as resolved. Please re-open this thread if this is a blocking issue.